docs/Tutorial_Matrix_Profile_Top_Ten_short.ipynb

```diff
@@ -0,0 +1,665 @@
+{
```

Comment:
<p>In unifrom scaling, data have to be located in same interval. Therefore scaled data should include non-original data point (which is at least on interpolated line). Its validity depends on domain and purpose of procesing. We can adjust scaled data by choosing interpolation function. I thought such kind of topics can be too detaild.</p>

Reply:
> In unifrom scaling, data have to be located in same interval. 

This is fine.

> Therefore scaled data should include non-original data point (which is at least on interpolated line). 

This is also fine and I do not dispute any of this.

> Its validity depends on domain and purpose of procesing. We can adjust scaled data by choosing interpolation function. I thought such kind of topics can be too detaild.

So, when I asked:

> Will I find the original (exact) values in the interpolated/fitted curve? Is it guaranteed?

I meant, given a dataset with points, say, `original = [0.0, 1.0, 2.0, 3.0, 4.0]` and we interpolate a dataset by adding a few points to `interpolated = [0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0]`. Notice that this contains new points (not a problem) but it also contains all of the original points as well. Is it always `True` that  `len(set(original).difference(set(interpolated))) == 0`?

Instead, if `interpolated = [0.0001, 0.5, 1.0001, 1.5, 2.0001, 2.5, 3.0001, 3.5, 4.0001]`, then `len(set(original).difference(set(interpolated))) == 5` (i.e., none of the original points are guaranteed to be found in `interpolated` then this may be undesirable.

Does that make sense?

Reply:
I understand what you meant. But I'm not still clear which point is undesirable.

> none of the original points are guaranteed to be found in `interpolated` 

if `interpolated = [0.0001, 0.5, 1.0001, 1.5, 2.0001, 2.5, 3.0001, 3.5, 4.0001]`, Indeed it doesn't include original points. But `interpolated`  should detect `original` because it has only very small error.
In aspect of finding motif, `interpolated` should be accurately "assumed sereis" rather than having exact orignal data point.
In calculating error, all point have same calculation weight for it . Majority point isn't on original data interval.  If I adjust `interpolated` to only have original data point, the data point not on `original` (like 0.5, 1.5) can be out of my "assummed line", and those point should cause unexpected error.

I feel at least in calculating error  `Interpolated ` should be adjusted to be "assummed line" in all point, but "assummed line" depends on the situation. So I don't know comprehensive way to check `Interpolated `.
In this `original = [0.0, 1.0, 2.0, 3.0, 4.0]`, as we can assume linearity.

The point you mention maybe is different from mine...

Do you think some kind of validation for `interpolated` is required?

Reply:
When I look at the paper, they perform `uniform scaling` in Matlab with:

```
T2 = T(1: 100/164: end); % Unofficial Matlab way to resample
```

Unfortunately, I do not have access to Matlab. When I use Octave (which is suppose to be "free-Matlab") with:

```
x = [1.01, 2.01, 3.01, 3.3, 3.5, 4.01, 5.01, 6.01]
x(1:100/150:end)  # stretch to 150%
```

It gives:

```
1.0100   2.0100   2.0100   3.0100   3.3000   3.3000   3.5000   4.0100  4.0100   5.0100   6.0100
```

Again, I don't know if this is the exact same as Matlab but, assuming that it is, this is what the paper is referring to for "uniform scaling". It appears that it is stretching out the length to 150% of the original length by repeating the original numbers. Notice that, unlike interpolation where completely new values are created, there are no new values in this time series (i.e., all values are derived from the original time input series). 

I'm not saying that this is great approach but we should try to recreate the paper where possible even if their approach is more naive. Otherwise, we'll end up with more questions that distract from the point. Additionally, we can always choose to add a "Bonus Section" at the end that offers suggestions for more "advance" scaling via interpolation.

Reply:
[This post](https://stackoverflow.com/questions/53904765/stretch-numpy-vector-to-arbitrary-length-without-interpolation) might offer some reasonable suggestions that are closer to the spirit of the Matlab code

Reply:
OK, I'm clear what you meant. But I don't know paper can repduce example with that code results. I have to check in "matlab" way. And why did author use unofficial resmapling code.. Anyway I'll try to get matlab enviroment.